server/chat-plugins/chat-monitor.ts

```diff
@@ -667,8 +667,65 @@ export const commands: ChatCommands = {
 		}
 		this.globalModlog(`ALLOWNAME`, target);
 	},
+	testfilter(target, room, user, connection, cmd) {
+		this.privatelyCheckCan('lock');
+		this.checkBroadcast(true);
+		if (room && ['staff', 'upperstaff'].includes(room.roomid)) this.runBroadcast();
+		let [message, monitorName] = target.split('|').map(x => x.trim());
+		if (!monitorName && cmd !== 'testfilter') {
+			monitorName = cmd.slice(4);
+			if (!Chat.monitors[monitorName] && Chat.monitors[monitorName.replace('filter', '')]) {
+				monitorName = monitorName.replace('filter', '');
+			}
+		}
+		// namefilter doesn't have a monitor function
+		if (!(monitorName && message) || !Chat.monitors[monitorName]?.monitor) {
+			return this.parse(`${this.cmdToken}help testfilter`);
+		}
+		const monitor = Chat.monitors[monitorName].monitor!;
+		const lcMessage = Chat.stripFormatting(message
+			.replace(/\u039d/g, 'N').toLowerCase()
+			// eslint-disable-next-line no-misleading-character-class
+			.replace(/[\u200b\u007F\u00AD\uDB40\uDC00\uDC21]/g, '')
+			.replace(/\u03bf/g, 'o')
+			.replace(/\u043e/g, 'o')
+			.replace(/\u0430/g, 'a')
+			.replace(/\u0435/g, 'e')
+			.replace(/\u039d/g, 'e'));
+		let htmlBoxMessage = ``;
+		for (const line of Chat.filterWords[monitorName]) {
+			const ret = monitor.call(this, line, room, user, message, lcMessage, true);
+			if (typeof ret === 'string') {
+				htmlBoxMessage = ret;
+				break;
+			}
+		}
+		if (htmlBoxMessage) {
+			return this.sendReplyBox(`${Chat.formatText(htmlBoxMessage, false, true)}`);
+		} else {
+			throw new Chat.ErrorMessage(
+				`"${message}" doesn't trigger any filters on the ${monitorName}${monitorName.endsWith('filter') ? '' : ' filter'}. Check spelling?`
+			);
+		}
+	},
+	testfilterhelp(target, room, user) {
+		this.privatelyCheckCan('lock');
+		this.checkBroadcast(true);
+		if (room && ['staff', 'upperstaff'].includes(room.roomid)) this.runBroadcast();
+		const monitorNames = Object.keys(Chat.monitors).filter(x => Chat.monitors[x].monitor);
+		this.sendReplyBox(
+			`<code>/test[monitorname]filter [test string]</code> OR <code>/testfilter [test string] | [monitorname]</code>:<br />Tests whether or not the provided test string would trigger the respective monitor.<br />` +
+			`All usable commands: <code>${monitorNames.map(x => `/test${x.replace('filter', '')}filter`).join('</code>, <code>')}</code><br />` +
+			`Can only be broadcast in Staff and Upper Staff. Requires: % @ &`
+		);
+	},
 };
 
+// Outside of `process.nextTick()` because it causes test to fail otherwise
+for (const list of Object.keys(Chat.monitors).filter(x => Chat.monitors[x].monitor)) {
```

Comment:
This needs to be in nextTick to ensure all monitors are loaded.

Reply:
it causes `npm test` to fail atrociously

```
 !

  0 passing (2ms)
  1 failing

  1) Uncaught error outside test suite:
     Uncaught TypeError: Cannot set property 'testautolockfilter' of undefined
      at C:\[redacted]\showdown\.server-dist\chat-plugins\chat-monitor.js:726:60
      at processTicksAndRejections (node:internal/process/task_queues:76:11)
```

Reply:
Attach it to Chat.commands instead? I would assume that would kill two birds with one stone (It's what I did locally when we discussed this).